### ISSUE TITLE
Longitudinal export should support repeatableQuestionGroups

### ISSUE THREAD

**Issue by sachsk** [avniproject/avni-server#547]
### Story
When repeatable question groups were added, csv exports did not support them. With this story, we will support longidutinal exports in repeatable question groups

### Acceptance criteria
- Longitudinal export will export repeatable question groups
- A spacer ```_``` and an index starting with 1 is used to to separate different different observations in the array. eg:  ```GroupConceptX_ConceptA_1, GroupConceptX_ConceptB_1, GroupConceptX_ConceptA_2, GroupConceptX_ConceptB_1``` for a form with 2 concepts ```ConceptA``` and ```ConceptB``` repeating twice
- Group concept name is added to the header
- Both old and new longitudinal exports are supported

### Tech notes
- Replace implementation of old export to new longitudinal export

### Testing notes
- Test with regular and repeatable question groups
- Test with empty (null) values, multiple encounters with diferent sizes
- Test with registration, enrolment and encounters
- Test both old and new versions
- It is expected to have new columns in the old export after this change. However, number of rows should not change. Old columns should still be available.

**Comment by vinayvenu** [comment]
@utkarshavni can you give some idea of the priority of this issue?

**Comment by petmongrels** [comment]
story notes
- why does multi-select field name provided as uuid?

**Comment by vinayvenu** [comment]
Request from @arjunk to potentially move this to 3.40 if possible.

**Comment by vinayvenu** [comment]
Minor issues


Move SubjectTypeBuild.java to test
We have two @RestControllerAdvice classes (AvniRestControllerAdvise and ErrorInterceptors)  that both deal with Errors differently. It is hard to follow the error response mechanism easily. Also, this one does not deal with bugsnag notifications. 

Why was logback-spring.xml moved to test?

EncounterRepository.getMaxEncounterCount() - why would this return a null? Both queries seem to return a count(*)

Same with ProgramEncounterRepository.getMaxProgramEncounterCount

**Comment by petmongrels** [comment]
1. done
2. done
3/4. these type of queries which has group by don't return 0, they return null - select count(id) from encounter_type et
    where et.name = 'zfdsfds'
group by et.name;

**Comment by himeshr** [comment]
Sample New Longitudinal export json query is attached here-in reference.

[DemandDistributionRepeatableQGRequest.json.zip](https://github.com/avniproject/avni-server/files/12358185/DemandDistributionRepeatableQGRequest.json.zip)


Reference SQL Commands:
---------------------------
-- #Query to fetch concept names and UUIDs
set role guatdbadmin;
select
    f.name  as FormName
    -- ,fm.entity_id
    -- ,fm.observations_type_entity_id
    -- ,fm.organisation_id
        ,
    feg.name,
    fe.name as "Form Element",
    c2.name as "Concept",
       c2.uuid as "ConceptUUID"
from form f
         inner join form_element_group feg on feg.form_id = f.id
         inner join form_element fe on fe.form_element_group_id = feg.id
         inner join concept c2 on fe.concept_id = c2.id
where f.name = 'Distribution Form' and not c2.is_voided and not fe.is_voided
order by
    f.name
       , feg.display_order asc
       , fe.display_order asc;

select * from encounter_type;
select * from subject_type;
